pymc/distributions/multivariate.py

```diff
@@ -2136,3 +2137,117 @@ def logp(value, mu, W, alpha, tau):
             at.all(alpha >= -1),
             tau > 0,
         )
+
+
+class StickBreakingWeightsRV(RandomVariable):
+    name = "stick_breaking_weights"
+    ndim_supp = 1
+    ndims_params = [0, 0]
+    dtype = "floatX"
+    _print_name = ("StickBreakingWeights", "\\operatorname{StickBreakingWeights}")
+
+    def __call__(self, alpha=1.0, K=10, size=None, **kwargs):
+        return super().__call__(alpha, K, size=size, **kwargs)
+
+    def _infer_shape(self, size, dist_params, param_shapes=None):
+        alpha, K = dist_params
+
+        size = tuple(size)
+
+        return size + (K + 1,)
+
+    @classmethod
+    def rng_fn(cls, rng, alpha, K, size):
+        if K < 0:
+            raise ValueError("K needs to be a positive integer.")
+
+        if np.ndim(alpha) > 0:
+            raise ValueError("The concentration parameter needs to be a scalar.")
+
+        if size is None:
+            size = (K,)
+        elif isinstance(size, int):
+            size = (size,) + (K,)
+        else:
+            size = tuple(size) + (K,)
+
+        betas = rng.beta(1, alpha, size=size)  # adapt this to vector alpha
+
+        sticks = np.concatenate(
+            (
+                np.ones(shape=(size[:-1] + (1,))),
+                np.cumprod(1 - betas[..., :-1], axis=-1),
+            ),
+            axis=-1,
+        )
+
+        weights = sticks * betas
+        weights = np.concatenate(
+            (weights, 1 - weights.sum(axis=-1)[..., np.newaxis]),
+            axis=-1,
+        )
+
+        return weights
+
+
+stickbreakingweights = StickBreakingWeightsRV()
+
+
+class StickBreakingWeights(Continuous):
+    rv_op = stickbreakingweights
+
+    def __new__(cls, name, *args, **kwargs):
+        kwargs.setdefault("transform", transforms.simplex)
+        return super().__new__(cls, name, *args, **kwargs)
+
+    @classmethod
+    def dist(cls, alpha, K, *args, **kwargs):
+        alpha = at.as_tensor_variable(floatX(alpha))
+        K = at.as_tensor_variable(intX(K))
+
+        if alpha.ndim > 0:
+            raise ValueError("alpha must be a scalar.")
+
+        if K.ndim > 0:
+            raise ValueError("K must be a scalar.")
+
+        assert_negative_support(alpha, "alpha", "StickBreakingWeights")
+        assert_negative_support(K, "K", "StickBreakingWeights")
+
+        return super().dist([alpha, K], **kwargs)
+
+    def get_moment(rv, size, alpha, K):
+        moment = (alpha / (1 + alpha)) ** at.arange(K + 1)
+        moment *= 1 / (1 + alpha)
+        if not rv_size_is_none(size):
+            moment = at.full(*size + (K + 1,), moment)
+        return moment
+
+    def logp(value, alpha, K):
+        if at.lt(K, 0).eval():
+            raise ValueError("K needs to be a positive integer.")
+
+        if at.lt(alpha, 0).eval():
+            raise ValueError("alpha needs to be a positive.")
+
+        if not at.allclose(value.sum(axis=-1), 1).eval():
+            raise ValueError("weights need to sum to 1.")
+
```

Comment:
Generally, I am using `.eval()` for some checks here, but I don't know if this is generalizable especially that inputs can be `np.ndarray` or Aesara tensors. Is there a better to do these checks?